CommonTools/Utils/interface/TrackSpecificColumns.h

```diff
@@ -0,0 +1,31 @@
+#ifndef CommonTools_Utils_TrackSpecificColumns_h
+#define CommonTools_Utils_TrackSpecificColumns_h
+
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "FWCore/SOA/interface/Column.h"
+
+namespace edm {
+
+  namespace soa {
+
+    namespace col {
+
+      SOA_DECLARE_COLUMN(PtError, float, "ptError");
+      SOA_DECLARE_COLUMN(D0, float, "d0");
+      SOA_DECLARE_COLUMN(NumberOfValidHits, int, "numberOfValidHits");
+      SOA_DECLARE_COLUMN(MissingInnerHits, int, "missingInnerHits");
+      SOA_DECLARE_COLUMN(Charge, int, "charge");
+
+    }  // namespace col
+
+    SOA_DECLARE_DEFAULT(col::PtError, ptError());
+    SOA_DECLARE_DEFAULT(col::D0, d0());
+    SOA_DECLARE_DEFAULT(col::NumberOfValidHits, numberOfValidHits());
+    SOA_DECLARE_DEFAULT(col::Charge, charge());
+    SOA_DECLARE_DEFAULT(col::MissingInnerHits, hitPattern().numberOfLostHits(reco::HitPattern::MISSING_INNER_HITS));
```

Comment:
this makes for much more aesthetically pleasing code.
The  DataFormats/TrackReco dependency I guess is unavoidable, or is it not?

Reply:
Yes because otherwise it doesn't know what `reco::HitPattern::MISSING_INNER_HITS` is...

Please don't test yet, I realized I made a mistake with the return type of this macro. Maybe it's also nice to look at without the macro. I was just worried that clang-format would split the overloaded functions over multiple lines, which would make it hard to read.

Reply:
Well, without a macro it reads a bit verbose, especially because we need the `template<>` if we don't want to introduce more external dependencies by making the arguments of these column fillers explicit. I leave it like that for today.

Reply:
I explicitly checked: header-checks will fail if you don't do the include from the Track DataFormats.

Reply:
Or maybe we could add a member function `missingInnerHits()` to TrackBase. In this way, the the uppercase column and lowercase filler pattern can also be kept consistently.

Reply:
sounds fine to me to add a member function.
I'd add both missingInner and missingOuter.